### PR TITLE
Change Score to accept double

### LINKED ISSUE #272: Modify Scores to accept floats
Common for exam scores to contain half mark.

### PR DESCRIPTION
**Pull request by delishad21**
Update functional code and test code for scores to accept double

Real world testing required to ensure that this functionality is bug free

Fixes #272

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Score.java [hunk 1]
```diff
@@ -3,28 +3,31 @@
...
 /**
  * Represents an Exam's score in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidScore(String)}
  */
 public class Score implements Comparable<Score> {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Scores should be numeric, and it should not be blank or less than zero.";
+            "Scores should be numeric, and it should not be blank or less than zero. "
+            + "Scores can have up to 2 decimal places.";
```
Comment (danielleloh) [inline src/main/java/seedu/address/model/person/Score.java:16]: reasonable change :) nice catch!

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 2]
```diff
@@ -108,4 +108,23 @@ public static boolean isNonZeroUnsignedInteger(String s) {
...
+
+    /**
+     * Returns true if {@code s} represents a non-negative unsigned double
+     * e.g. 0.0, 1.0, 2.0, 3.0, ..., {@code Double.MAX_VALUE} <br>
+     * Will return false for any other non-null string input
+     * e.g. empty string, "-1.0", "+1.0", and " 2.0 " (untrimmed),
+     * "3.0 0.0" (contains whitespace), "1.0 a" (contains letters)
+     * @throws NullPointerException if {@code s} is null.
+     */
+    public static boolean isNonNegativeUnsignedDouble(String s) {
```
Comment (jayllo-c) [inline src/main/java/seedu/address/commons/util/StringUtil.java:120]: Nice job upgrading StringUtil to abstract the verification of string values

File: src/main/java/seedu/address/model/person/Score.java [hunk 3]
```diff
@@ -33,17 +36,23 @@ public Score(int score) {
     /**
      * Returns true if a given string is a valid score.
      */
-    public static boolean isValidScore(int test) {
-        return test >= 0;
+    public static boolean isValidScore(double test) {
+        String str = Double.toString(test);
+        return str.matches(VALIDATION_REGEX);
```
Comment (jayllo-c) [inline src/main/java/seedu/address/model/person/Score.java:41]: Nice use of methods in String

**Review by jayllo-c — COMMENTED** [review]
Nice job implementing the changes to have the score stored as a double

File: src/main/java/seedu/address/commons/util/StringUtil.java (hunk 2 above)
Comment (Zer0Legion) [reply src/main/java/seedu/address/commons/util/StringUtil.java:120]: Good abstraction

File: src/main/java/seedu/address/logic/commands/ImportExamCommand.java
Comment (Zer0Legion) [inline src/main/java/seedu/address/logic/commands/ImportExamCommand.java:1]: Thanks for updating this for me!

**Pull request merged**